File: hotelaria-frontend/src/app/shared/enums/TiposProdutoEnum.ts
export class TiposProdutoEnum {
    static COMIDA = new TiposProdutoEnum("Comida");
    static BEBIDA = new TiposProdutoEnum("Bebida");
    static SERVICO_ADICIONAL = new TiposProdutoEnum("Serviço Adicional");

    private constructor(private descricao: string) {}

    getDescricao(): string {
        return this.descricao;
    }
}

File: Hotelaria/src/main/java/com/dolts/hotelaria/enums/TiposProdutoEnum.java
package com.dolts.hotelaria.enums;

import lombok.Getter;

@Getter
public enum TiposProdutoEnum {
    COMIDA("Comida"),
    BEBIDA("Bebida"),
    SERVICO_ADICIONAL("Serviço Adicional");

    private final String descricao;

    TiposProdutoEnum(String descricao) {
        this.descricao = descricao;
    }

}

